contracts/BuyBackBurner.sol

```diff
@@ -117,10 +118,23 @@ contract BuyBackBurner {
         require(pool != address(0), "Pool does not exist");
 
         // Get current pool reserves
-        (uint160 sqrtPriceX96, , , , , , ) = IUniswapV3(pool).slot0();
+        (uint160 sqrtPriceX96, , uint16 observationCardinality, , , , ) = IUniswapV3(pool).slot0();
+        // Check observation cardinality
+        if (observationCardinality < 2) {
+            // Increase observation cardinality to get more accurate twap
+            IUniswapV3(pool).increaseObservationCardinalityNext(60);
+            return;
+        }
+
+        // Check if the pool has sufficient observation history
+        (uint32 oldestTimestamp, , , ) = IUniswapV3(pool).observations(0);
```

Comment:
If the latest observation was happening earlier than the 1800 seconds window, this definitely means that there was no activity and the instant price is correct.